aries_vcx_core/src/wallet/askar/askar_wallet.rs

```diff
@@ -0,0 +1,289 @@
+use std::collections::HashMap;
+
+use aries_askar::{
+    entry::EntryTag,
+    kms::{KeyAlg, KeyEntry, LocalKey},
+    PassKey, Session, Store, StoreKeyMethod,
+};
+use async_trait::async_trait;
+
+use super::rng_method::RngMethod;
+#[cfg(feature = "vdrtools_wallet")]
+use crate::WalletHandle;
+use crate::{
+    errors::error::{AriesVcxCoreError, AriesVcxCoreErrorKind, VcxCoreResult},
+    utils::async_fn_iterator::AsyncFnIterator,
+    wallet::{base_wallet::BaseWallet, structs_io::UnpackMessageOutput},
+};
+
+#[derive(Debug)]
+pub struct AskarWallet {
+    pub backend: Store,
+    profile: Option<String>,
+}
+
+impl AskarWallet {
+    pub async fn create(
+        db_url: &str,
+        key_method: StoreKeyMethod,
+        pass_key: PassKey<'_>,
+        recreate: bool,
+        profile: Option<String>,
+    ) -> Result<Self, AriesVcxCoreError> {
+        let backend =
+            Store::provision(db_url, key_method, pass_key, profile.clone(), recreate).await?;
+
+        Ok(Self { backend, profile })
+    }
+
+    pub async fn open(
+        db_url: &str,
+        key_method: Option<StoreKeyMethod>,
+        pass_key: PassKey<'_>,
+        profile: Option<String>,
+    ) -> Result<Self, AriesVcxCoreError> {
+        let backend = Store::open(db_url, key_method, pass_key, profile.clone()).await?;
+
+        Ok(Self { backend, profile })
+    }
+
+    pub async fn create_key(
+        &self,
+        name: &str,
+        alg: KeyAlg,
+        seed: &str,
+        rng_method: RngMethod,
+        metadata: Option<&str>,
+        tags: Option<&[EntryTag]>,
+    ) -> Result<(), AriesVcxCoreError> {
+        let mut session = self.backend.session(self.profile.clone()).await?;
+
+        let key = LocalKey::from_seed(alg, seed.as_bytes(), rng_method.into())?;
+        Ok(session.insert_key(name, &key, metadata, tags, None).await?)
+    }
+
+    pub async fn create_did(
+        &self,
+        key_name: &str,
+        profile: Option<String>,
+    ) -> Result<(), AriesVcxCoreError> {
+        let mut session = self.backend.session(profile).await?;
+
+        let key_entry = self.fetch_key_entry(&mut session, key_name).await?;
+
+        let local_key = key_entry.load_local_key()?;
+
+        let did_bytes = &local_key.to_public_bytes()?[0..16];
```

Comment:
`create_and_store_my_did` from `BaseWallet` always derives did from verkey, but it might be nice to allow did to be passed in here as an argument.